From af10552a3670dda7d2e77a8ebc390db8b517e92c Mon Sep 17 00:00:00 2001
From: Vincent Davis Jr <vince@underview.tech>
Date: Tue, 17 Jan 2023 19:14:07 -0600
Subject: [PATCH] am335x-customboneblack.dts: add new device tree

Add device tree for uart devices and configure
pinmux information on the am3358 ARM Cortex A8
processor so that pins 21,22 (uart 2) and pins
11,13 (uart 4) on Expansion Header P9 are
utilized as uart pins.

Signed-off-by: Vincent Davis Jr <vince@underview.tech>
---
 arch/arm/boot/dts/am335x-customboneblack.dts | 36 ++++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 arch/arm/boot/dts/am335x-customboneblack.dts

diff --git a/arch/arm/boot/dts/am335x-customboneblack.dts b/arch/arm/boot/dts/am335x-customboneblack.dts
new file mode 100644
index 000000000000..64ac1a3574bb
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-customboneblack.dts
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/dts-v1/;
+
+#include "am335x-boneblack.dts"
+
+&am33xx_pinmux {
+	/* Pins 21 (TX) and 22 (RX) of connector P9 */
+	uart2_pins: uart2_pins {
+		pinctrl-single,pins = <
+			AM33XX_PADCONF(AM335X_PIN_SPI0_SCLK, PIN_INPUT_PULLUP, MUX_MODE1) /* (A17) spi0_sclk -> uart2_rxd */
+			AM33XX_PADCONF(AM335X_PIN_SPI0_D0, PIN_OUTPUT, MUX_MODE1)         /* (B17) spi0_d0 -> uart2_txd */
+		>;
+	};
+
+	/* Pins 11 (RX) and 13 (TX) of connector P9 */
+	uart4_pins: uart4_pins {
+		pinctrl-single,pins = <
+			AM33XX_PADCONF(AM335X_PIN_GPMC_WAIT0, PIN_INPUT_PULLUP, MUX_MODE6) /* (T17) gpmc_wait0 -> uart4_rxd */
+			AM33XX_PADCONF(AM335X_PIN_GPMC_WPN, PIN_OUTPUT, MUX_MODE6)         /* (U17) gpmc_wpn -> uart4_txd */
+		>;
+	};
+};
+
+&uart2 {
+	compatible = "underview,serial-uart";
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2_pins>;
+};
+
+&uart4 {
+	compatible = "underview,serial-uart";
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart4_pins>;
+};
-- 
2.34.1

